lib/spack/spack/mirror.py

```diff
@@ -49,8 +49,9 @@ def mirror_archive_filename(spec, fetcher, resourceId=None):
     if not spec.version.concrete:
         raise ValueError("mirror.path requires spec with concrete version.")
 
+    name = None
     if isinstance(fetcher, fs.URLFetchStrategy):
-        if fetcher.expand_archive:
+        if fetcher.fetch:
```

Comment:
```fetch``` is a method for FetchStrategy so I presume it will always be defined. I don't think this is the appropriate check. Taking a quick look through our unit tests, it appears that we don't have anything that checks a package that should not be expanded, otherwise I think such a test would have caught this.